src/components/checkbox/checkbox.test.js

```diff
@@ -0,0 +1,549 @@
+  });
+
+  it("should render Checkbox component with info ", () => {
+    CypressMountWithProviders(<CheckboxComponent info />);
+
+    checkboxSvg().should("have.css", "border-bottom-color", "rgb(0, 96, 167)");
+  });
+
+  it("should render Checkbox component with error message ", () => {
+    CypressMountWithProviders(<CheckboxComponent error="Error has occurred" />);
+
+    checkboxIcon()
+      .should("have.attr", "aria-label", "Error has occurred")
+      .should("have.attr", "data-element", "error");
+  });
+
+  it("should render Checkbox component with warning message", () => {
+    CypressMountWithProviders(
+      <CheckboxComponent warning="Warning has occurred" />
+    );
+
+    checkboxIcon()
+      .should("have.attr", "aria-label", "Warning has occurred")
+      .should("have.attr", "data-element", "warning");
+  });
+
+  it("should render Checkbox component with info message", () => {
+    CypressMountWithProviders(<CheckboxComponent info="Info has occurred" />);
+
+    checkboxIcon()
+      .should("have.attr", "aria-label", "Info has occurred")
+      .should("have.attr", "data-element", "info");
+  });
+
+  it.each([
+    [true, "0"],
+    [false, "1"],
+  ])(
+    "should render Checkbox with reverse prop set to %s",
+    (reverseValue, position) => {
+      CypressMountWithProviders(
+        <CheckboxComponent label="Checkbox Label" reverse={reverseValue} />
+      );
+
+      checkboxComponent()
+        .find("div:nth-child(1) > div > div:nth-child(1)")
+        .children()
+        .eq(position)
+        .should("have.text", "Checkbox Label");
+    }
+  );
+
+  it.each(["bottom", "left", "right", "top"])(
+    "should render CheckboxComponent component with tooltip positioned to the %s",
+    (position) => {
+      CypressMountWithProviders(
+        <CheckboxComponent
+          labelHelp="Tooltip info"
+          tooltipPosition={position}
+        />
+      );
+      checkboxIcon().trigger("mouseover");
+      tooltipPreview()
+        .should("have.text", "Tooltip info")
+        .should("have.attr", "data-placement", `${position}`);
+    }
+  );
+
+  it("should render Checkbox component with tick color set to rgb(0, 0, 0)", () => {
+    CypressMountWithProviders(<CheckboxComponent checked />);
+    checkboxRole()
+      .should("be.checked")
+      .should("have.css", "color", "rgb(0, 0, 0)");
+  });
+
+  describe("should render CheckBox component and check events", () => {
+    let callback;
+
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it("should call onBlur callback when a blur event is triggered", () => {
+      CypressMountWithProviders(<CheckboxComponent onBlur={callback} />);
+
+      checkboxRole()
+        .focus()
+        .blur()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onChange callback when an onChange event is triggered", () => {
```

Comment:
we need to add one more test to check the this event will be triggered for `uncheck()` too

Reply:
added